pkg/util/validation/limits.go

```diff
@@ -280,6 +281,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&l.OTelMetricSuffixesEnabled, "distributor.otel-metric-suffixes-enabled", false, "Whether to enable automatic suffixes to names of metrics ingested through OTLP.")
 	f.BoolVar(&l.OTelCreatedTimestampZeroIngestionEnabled, "distributor.otel-created-timestamp-zero-ingestion-enabled", false, "Whether to enable translation of OTel start timestamps to Prometheus zero samples in the OTLP endpoint.")
 	f.Var(&l.PromoteOTelResourceAttributes, "distributor.otel-promote-resource-attributes", "Optionally specify OTel resource attributes to promote to labels.")
+	f.Var(&l.DistributorArtificialIngestionDelay, "distributor.artificial-ingestion-latency", "Target ingestion delay. If set to a non-zero value, the distributor will artificially delay ingestion time-frame by the specified duration by computing the difference between actual ingestion and the target. There is no delay on actual ingestion of samples, it is only the response back to the client.")
```

Comment:
Can you rename this to remove the "Distributor" prefix of the configuration setting and variable? It's already named to indicate it's part of the ingestion path and we don't do this with other settings.

Reply:
Are you certain? I've introduced a few recently in the ruler and they all have them - I thought it was more of a "introduce it in the new ones so that we maybe one day reach consistency". We have so many flags in this struct that I think the prefix is beneficial.

Reply:
I don't feel strong on this, but an option to follow nick's suggestion is to rename the field to `ingestion_artificial_delay` (and CLI to `-distributor.ingestion-artificial-delay`). It would be consistent with `IngestionTenantShardSize`.

Reply:
I prefer Marco's suggestion to use `ingestion_artificial_delay`. I don't like including "distributor" in the setting name since it's an implementation detail.

Reply:
Done in https://github.com/grafana/mimir/pull/10107/commits/945dfb844564683d41a87fd4f290e626007805cd